src/api/firebase.js

```diff
@@ -195,23 +195,44 @@ export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
+		purchaseInterval: daysUntilNextPurchase,
 	});
 }
 
 export async function updateItem(
 	listPath,
-	{ itemId, totalPurchases, dateLastPurchased },
+	{ itemId, totalPurchases, dateLastPurchased, purchaseInterval, dateCreated },
 ) {
 	/**
 	 * TODO: Fill this out so that it uses the correct Firestore function
 	 * to update an existing item. You'll need to figure out what arguments
 	 * this function must accept!
 	 */
+
 	const itemDocRef = doc(db, `${listPath}/items`, itemId);
 
-	await updateDoc(itemDocRef, {
+	let daysSinceLastPurchase;
+
+	if (dateLastPurchased) {
+		daysSinceLastPurchase = getDaysBetweenDates(dateLastPurchased);
+	} else {
+		daysSinceLastPurchase = getDaysBetweenDates(dateCreated);
+	}
+
+	const daysUntilNextPurchase = calculateEstimate(
+		purchaseInterval,
+		daysSinceLastPurchase,
 		totalPurchases,
-		dateLastPurchased,
+	);
+
+	purchaseInterval = daysUntilNextPurchase;
+
+	const dateNextPurchased = getFutureDate(daysUntilNextPurchase);
```

Comment:
I think your logic makes sense here, however it is a little bit hard to follow (in nature due to the calculation itself being tricky. I would consider adding comments to each line to make sure it is clear what each line is doing. 

This also feels like quite a big block of code. I would consider trying to separate out anything that is not the purpose of the `updateItem` function  (to update the database with the item's updated data), and seeing where else I could put this calculating dates logic - maybe in the list component itself, or another util function? You will then need to change the arguments for this function. This way we can keep this main function less noisy.